tasks/main.yml

```diff
@@ -19,6 +35,26 @@
     - not network_packages is subset(ansible_facts.packages.keys())
   register: __network_package_install
 
+- name: Install NetworkManager and nmstate when using network_state variable
+  package:
+    name:
+      - NetworkManager
+      - nmstate
+    state: present
+  when: network_state is defined
+
+- name: Install nmstate-devel when using network_state variable
+  package:
+    name:
+      - nmstate-devel
+    state: present
+  when:
+    - network_state is defined
+    - ansible_distribution == 'Fedora' and
+      ansible_distribution_major_version | int > 34 or
+      ansible_distribution != 'Fedora' and
+      ansible_distribution_major_version | int > 8
```

Comment:
Shouldn´t this install NetworkManager too? As the test above is doing? Also the check on nmstate is also nice to have.

Reply:
NM is always installed in the previous task.

Reply:
I added the system version check on the nmstate.